src/Plugin/EntityBrowser/Widget/View.php

```diff
@@ -199,11 +165,11 @@ public static function processCheckbox(&$element, FormStateInterface $form_state
    * {@inheritdoc}
    */
   public function submit(array &$element, array &$form, FormStateInterface $form_state) {
-    $selected_rows = array_keys(array_filter($form_state->getValue('entity_browser_select')));
+    $selected_rows = array_values(array_filter($form_state->getUserInput()['entity_browser_select']));
     $entities = [];
-    $ids = $form_state->get('view_widget_rows');
     foreach ($selected_rows as $row) {
-      $entities[] = $this->entityManager->getStorage($ids[$row]['type'])->load($ids[$row]['id']);
+      list($type, $id) = explode(':', Html::escape($row));
```

Comment:
I don't think Html::escape() helps in any way. As the name says, that's meant to be used when displaying something on an HTML page. If you want to test something, then make sure you have exactly one : in here and type is a valid entity type. id is validated by load().